indico/modules/auth/forms.py

```diff
@@ -37,11 +37,42 @@ def _check_existing_email(form, field):
         raise ValidationError(_('This email address is already in use.'))
 
 
+def _check_existing_username(form, field):
+    if Identity.find(provider='indico', identifier=field.data).count():
+        raise ValidationError(_('This username is already in use.'))
+
+
 class LocalLoginForm(IndicoForm):
     identifier = StringField(_('Username'), [DataRequired()], filters=[_tolower])
     password = PasswordField(_('Password'), [DataRequired()])
 
 
+class LocalLoginAddForm(IndicoForm):
+    username = StringField(_('Username'), [DataRequired(), _check_existing_username], filters=[_tolower])
+    password = PasswordField(_('Password'), [DataRequired(), Length(min=5)])
+    confirm_password = PasswordField(_('Confirm password'), [DataRequired(), ConfirmPassword('password')])
+
+
+class LocalLoginEditForm(IndicoForm):
+    username = StringField(_('Username'), [DataRequired()], filters=[_tolower])
+    password = PasswordField(_('Current password'), [DataRequired()])
+    new_password = PasswordField(_('New password'), [Optional(), Length(min=5)])
+    confirm_new_password = PasswordField(_('Confirm password'), [ConfirmPassword('new_password')])
+
+    def __init__(self, *args, **kwargs):
+        self.identity = kwargs.pop('identity', None)
+        super(LocalLoginEditForm, self).__init__(*args, **kwargs)
+
+    def validate_password(self, field):
+        if field.data != self.identity.password:
+            raise ValidationError(_("Wrong current password"))
+
+    def validate_username(self, field):
+        query = Identity.find(Identity.identifier != self.identity.identifier, provider='indico', identifier=field.data)
```

Comment:
Mostly a matter of taste, but I'm not a big fan of mixing args and kwargs in the same find call